### DISCUSSION TITLE
How to use DTO to "return relations"

### DISCUSSION THREAD

**Discussion by YuriAlessandro** [litestar-org/litestar#710] [Q&A] [answered]
Hello. I'm using the Tortoise plugin and I want to know how can I return the values of a relation in the controller.

If i have a relation between two models, for example:

```python
class User(Model):
  id = fields.IntField(pk=True)
  name = fields.CharField(max_length=255)

class Photo(Model):
  id = fields.IntField(pk=True)
  number = fields.CharField(max_length=255)
  status = fields.CharField(max_length=255)

  user = fields.ForeignKeyField('models.User', related_name='users')

UserDTO = dto_factory("UserDTO", User)
```

In the controller i'm doing:

```python
@get()
async def list_users() -> list[UserDTO]:
  users = await User.all()
  return cast("list[UserDTO]", users )
```

But this returns only `{ id: 1, number: 0, status: 'completed'}`. How can I include the return of "users" in the model User?

**Comment by jtraub** [comment, marked as answer]
It doesn't show up there because when DTO factories are created Tortoise is not aware of relations. Internally the plugin uses `tortoise.contrib.pydantic.pydantic_model_creator` which can work with relations if models have been initialized before its invocation.


Please refer to [Tortoise docs](https://tortoise-orm.readthedocs.io/en/latest/contrib/pydantic.html#relations-early-init)
